### PR TITLE
Add pytest operator testing and LXD group creation 

### PR DESCRIPTION
**Pull request by DomFleischmann**
The following tests have been added:

- Building a charm with charmcraft and pytest-operator to make sure
the environment is setup correctly.

- Deploying Ubuntu and Minio with pytest operator instead of the juju
cli.

These tests are executed with tox.

Additionally the LXD user group will be created during setup, this is
needed for charmcraft to work correctly

### CODE REVIEW

File: dist/bootstrap/index.js [hunk 1]
```diff
@@ -7,14 +7,27 @@ module.exports =
 /***/ (function(__unused_webpack_module, exports, __nccwpck_require__) {
```
Comment (knkski) [inline dist/bootstrap/index.js:7]: Any idea why there's so many changes for such a simple TS change?

**Review by knkski — APPROVED** [review]
LGTM

File: dist/bootstrap/index.js (hunk 1 above)
Comment (johnsca) [reply dist/bootstrap/index.js:7]: My guess would be the update to `actions/core` but the compilation step is a bit opaque, regardless.

**Review by johnsca — APPROVED** [review]
I'm not a huge fan of the circular dependency this introduces in the testing; ideally the tests in this repo wouldn't depend on pytest-operator since it depends on this and doing so opens up the possibility of mutually dependent PRs, exactly like we have here, where we can't verify that either are actually working until merging one or both with failing tests. That said, I also recognize that we don't want to reinvent the wheel for the tests in this repo and they two components are intended to work together, so it's probably not worth worrying about.

So, slightly reluctant +1

**Pull request merged**